login/service_whitebox_test.go

```diff
@@ -0,0 +1,151 @@
+package login
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/token"
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+var db *gorm.DB
+var loginService *gitHubOAuth
+
+// Github doesn't allow commiting actual tokens no matter how
+// less privleges the token has.
+var camouflagedAccessToken = "751e16a8b39c0985066-AccessToken-4871777f2c13b32be8550"
+var actualToken = strings.Split(camouflagedAccessToken, "-AccessToken-")[0] + strings.Split(camouflagedAccessToken, "-AccessToken-")[1]
+
+func setup() {
```

Comment:
Is it still failing for the name `TestMain` ? 
Because it is adding the burden of 
```
if db == nil || loginService == nil {
	setup()
}
```
in every test !